File: C++/3_Insertion Sort/Insertion Sort.cpp
#include <iostream>
using namespace std;

void swap(int *a, int *b){
	int temp = *a;
	*a = *b;
	*b = temp;
}

void insertionSort(int arr[], int n){
	int i, j;
	for(i = 1; i < n; i++){
		j = i;
		while(j > 0 && arr[j-1] > arr[j]){
			swap(&arr[j], &arr[j-1]);
			j--;
		}
	}
}

void printArray(int arr[], int n){
	for(int i = 0; i < n; i++){
		cout<<arr[i]<<" ";
	}
}

int main() {
	int arr[] = {5, 2, 42, 6, 1, 3, 2};
	int n = sizeof(arr)/sizeof(arr[0]);
	insertionSort(arr, n);
	printArray(arr, n);
	return 0;
}

File: Javascript/3_Insertion Sort/Insertion Sort.js
function insertionSort(array){
    var j;
    for(var k = 1; k < array.length; k++){
        j = k;
        while(j > 0 && array[j-1] > array[j]){
            var temp = array[j];
            array[j] = array[j-1];
            array[j-1] = temp;
            j--;
        }
    }
}

var array = [12, 11, 13, 5, 6]
insertionSort(array);
console.log(array);